include/crocoddyl/multibody/contacts/contact-2d.hxx

```diff
@@ -0,0 +1,121 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+namespace crocoddyl {
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::ContactModel2DTpl(boost::shared_ptr<StateMultibody> state, const FrameTranslation& xref,
+                                             const std::size_t& nu, const Vector2s& gains)
+    : Base(state, 2, nu), xref_(xref), gains_(gains) {}
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::ContactModel2DTpl(boost::shared_ptr<StateMultibody> state, const FrameTranslation& xref,
+                                             const Vector2s& gains)
+    : Base(state, 2), xref_(xref), gains_(gains) {}
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::~ContactModel2DTpl() {}
+
+template <typename Scalar>
+void ContactModel2DTpl<Scalar>::calc(const boost::shared_ptr<ContactDataAbstract>& data,
+                                     const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+  pinocchio::updateFramePlacement(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->v = pinocchio::getFrameVelocity(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->vw = d->v.angular();
+  d->vv = d->v.linear();
+
+  pinocchio::getFrameJacobian(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame, pinocchio::LOCAL, d->fJf);
+  d->Jc.row(0) = d->fJf.row(0);
+  d->Jc.row(1) = d->fJf.row(2);
+
+  d->a = pinocchio::getFrameAcceleration(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->a0[0] = d->a.linear()[0] + d->vw.cross(d->vv)[0];
+  d->a0[1] = d->a.linear()[2] + d->vw.cross(d->vv)[2];
```

Comment:
In these lines, you are twice doing vector multiplication, computing the same products, and then using only part of it.
What you can do is to say 
```
d->a0[0] = d->a.linear()[0] + d->vw[1]*d->vv[2] - d->vw[2]*d->vv[1];
d->a0[0] = d->a.linear()[0] + d->vw[0]*d->vv[1] - d->vw[1]*d->vv[0];
```
(check the indices)